File: arrays/045-jump-game-ii/JumpGameII.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        far = 0
        jumps = 0
        end = 0
        for i in range(len(nums)-1):
            far = max(far, i + nums[i])
            if i == end:
                jumps+=1
                end = far
        return jumps

File: arrays/045-jump-game-ii/JumpGameII.java
class Solution {
    public int jump(int[] nums) {
        int far = 0, end = 0, jumps = 0;
        for(int i =0 ;i<nums.length-1;i++){
            far = Math.max(far, i + nums[i]);
            if(i == end){
                jumps++;
                end = far;
            }
        }
        return jumps;
    }
}